### PR TITLE
Fix rpmverify and rpmverifyfile unit tests on suse based systems

### PR DESCRIPTION
**Pull request by rfrohl**
This a suggestion for how a solution for #1258 could look. This worked in my environment, but I am not sure if this would be an acceptable change.

Please let me know if changes to the unit test are interesting in general. I have a few more, but I currently lack an option to test them on anything that is not Suse based.

### CODE REVIEW

**Comment by jan-cerny** [comment]
@openscap-ci ok to test

File: tests/probes/rpmverifyfile/test_probes_rpmverifyfile_older.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
         <lin-def:version operation="pattern match"/>
         <lin-def:release operation="pattern match"/>
         <lin-def:arch operation="pattern match"/>
-        <lin-def:filepath>/etc/passwd</lin-def:filepath>
+        <lin-def:filepath>/etc/securetty</lin-def:filepath>
```
[the remark this answers is not in the document]
Comment (rfrohl) [reply tests/probes/rpmverifyfile/test_probes_rpmverifyfile_older.xml]: I will check for an alternative and send an update.

[the remark this answers is not in the document]
Comment (rfrohl) [reply tests/probes/rpmverifyfile/test_probes_rpmverifyfile_older.xml]: would '/etc/os-release' be viable alternative ?

For Suse it is also present in the build chroot.

[the remark this answers is not in the document]
Comment (yuumasato) [reply tests/probes/rpmverifyfile/test_probes_rpmverifyfile_older.xml]: I think that the purpose of the test is not to test specifically `/etc/passwd` or its equivalent in other systems.
The test is looking to scan a path, and check the actual paths that the probe has looked into.
We could change this test to look into a path that is common in both Fedora and Suse.

[the remark this answers is not in the document]
Comment (yuumasato) [reply tests/probes/rpmverifyfile/test_probes_rpmverifyfile_older.xml]: For some reason I haven't seen the newer comments before commenting.

[the remark this answers is not in the document]
Comment (rfrohl) [reply tests/probes/rpmverifyfile/test_probes_rpmverifyfile_older.xml]: As far as I understand the test the file/directory needs to be owned by an rpm. The problem is specifically on suse systems that '/etc/passwd' does not belong to any rpm.

'/etc/os-release' seems to be an alternative. For Suse it is also present in the chroot during build and it exists on Fedora.

[the remark this answers is not in the document]
Comment (rfrohl) [reply tests/probes/rpmverifyfile/test_probes_rpmverifyfile_older.xml]: I pushed a change for this so there is some feedback from the CI

[the remark this answers is not in the document]
Comment (rfrohl) [reply tests/probes/rpmverifyfile/test_probes_rpmverifyfile_older.xml]: Looks like `/etc/os-release` does not work for rhel7, will look for an alternative for the alternative next week

**Comment by jan-cerny** [comment]
@openscap-ci test this please

**Comment by jan-cerny** [comment]
Hi @rfrohl.

The test `test_not_equals_operation` is driven by  `tests/probes/rpmverify/test_not_equals_operation.sh`. From that we can see that it basically runs `oscap oval eval test_not_equals_operation.xml`. Then runs some XPath queries on the results file to check if `oscap` has generated expected results. When I take a look to the XML file, I can guess what is the feature that is being tested. It looks like it tries to test the comparison of RPM package names. However, it's done in this very convoluted way (describing the current upstream, without your changes):
1. it evaluates `<rpmverify_object id="oval:x:obj:1">` which returns an object describing rpm package which owns `/` path. On my Fedora machine the result `name` is `filesystem`.
2. it saves the result of evaluation of   `"oval:x:obj:1"` to local variable `"oval:x:var:1"` - therefore `"oval:x:var:1"` is set to `filesystem`.
3. it evaluates `<rpmverify_object id="oval:x:obj:2">` which checks for rpm package(s) which own `/` or `/etc/passwd` path AND it's name is not equal to value of `"oval:x:var:1"`. The `name` of the owner of `/` is `filesystem`, the owner of `/etc/passwd` is `setup`, but it must be not equal to `"oval:x:var:1"`, which means it must be not equal to `filesystem`, it removes away the `filesystem` package from the result set and result is only the `setup`.

> My understanding is that the test fails on rhel7 because /etc is owned by two rpms.

I think that is a correct observation. 

> With my limit understanding I was wondering if 'matching' the expected number of results would be enough to satisfy the test. This could be achieved by using /root instead of /etc.

Not exactly. You need to find 2 different paths that: 
* are always owned by a package
* first path is always owned by a different package than the second path.
Then you put on of them to `oval:x:obj:1` and both of them to the regex in `oval:x:obj:2`.
But yes,  `/root` and `/etc/os-release` look like a promising couple, it works on my Fedora.

> What I am unsure about is how the rpmverify_object knows how many results to expect. At least from the comments I get the impression that a specific number of results is expected. 

The `rpmverify_object` doesn't know anything about the expected amount of results. I think that was the author of the test that assumed the object will always return 2 objects, because it happened on his operating system. I think he just haven't realized that this is not always the case.

> One thing I am still unsure about is if the rhel7 and fedora container used for verifying this information resembles the systems used by the CI.

The Jenkins CI uses virtual machines. By using containers you can simulate the CI environment. But there will be still some differences (eg. systemd related tests will not work properly). My impression is that for the RPM related rules it should work the same way in containers.

TBH  this is the thing I don't like about our test suite: it heavily depends on system and environment. I think that the features that are tested by this test could be tested in a less convoluted way.

**Comment by rfrohl** [comment]
@jan-cerny: thanks a lot for taking the time to explain it.

**Pull request merged**